core/src/main/scala/kafka/server/BrokerBlockingSender.scala

```diff
@@ -139,17 +145,42 @@ class RemoteBrokerBlockingSender(sourceBroker: BrokerEndPoint,
                                  time: Time,
                                  fetcherId: Int,
                                  clientId: String,
-                                 logContext: LogContext) extends BlockingSend {
+                                 logContext: LogContext,
+                                 clusterLinkProperties: Properties = new Properties()) extends BlockingSend {
 
   private val sourceNode = new Node(sourceBroker.id, sourceBroker.host, sourceBroker.port)
   private val socketTimeout: Int = brokerConfig.replicaSocketTimeoutMs
 
   private val (networkClient, reconfigurableChannelBuilder) = {
-
-    val channelBuilder = ClientUtils.createChannelBuilder(brokerConfig, time, logContext)
+    // Configure network client with cluster link properties
+    val securityProtocol = Option(clusterLinkProperties.getProperty(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG))
+      .map(SecurityProtocol.forName)
+      .getOrElse(SecurityProtocol.PLAINTEXT)
+    val saslMechanism = Option(clusterLinkProperties.getProperty(SaslConfigs.SASL_MECHANISM))
+      .getOrElse("PLAIN")
+    val saslJaasConfig = clusterLinkProperties.getProperty(SaslConfigs.SASL_JAAS_CONFIG)
+    if (saslJaasConfig != null) {
+      clusterLinkProperties.put(SaslConfigs.SASL_JAAS_CONFIG, new Password(saslJaasConfig))
+    }
```

Comment:
If we `ClusterLinkConfigs`, which contains the definition of configs, like `AdminClientConfigs` did, it will be parsed as the expected type like `Password` type here automatically.

Reply:
Yes, fixed.